models/utils.models.js

```diff
@@ -0,0 +1,53 @@
+function queryFormat(query) {
+  let sqlString = "";
+  let valueArr = [];
+  const validOrder = ["ASC", "DESC"];
+  const validSortBy = ["created_at", "comment_count", "article_id", "votes"];
+  const validQuery = ["topic", "order", "sort_by"];
+
+  //BAD REQUEST
+  const keys = Object.keys(query);
+  const validateKeyBoolean = keys.some((query) => validQuery.includes(query));
+
+  if (!validateKeyBoolean && keys.length > 0) {
+    return {
+      validationErr: Promise.reject({ status: 400, msg: "Bad Request" }),
+    };
+  }
+
+  if (keys.includes("order") || keys.includes("sort_by")) {
+    const values = [query.order, query.sort_by];
+    const combValidValues = validOrder.concat(validSortBy);
+    const validateValueBoolean = values.some((value) =>
+      combValidValues.includes(value)
+    );
+
+    if (!validateValueBoolean) {
+      return {
+        validationErr: Promise.reject({ status: 400, msg: "Bad Request" }),
+      };
+    }
+  }
+
+  //SQL STRING ADD-ON
+  if (query.topic) {
+    sqlString += `WHERE articles.topic = $1`;
+    valueArr.push(query.topic);
+  }
+
+  sqlString += "GROUP BY 1,2,3,4,5,6,7";
```

Comment:
I am unsure as to why you are grouping by numbers but your GROUP BY just needs to be a unique value. This should still work if we grouped by article.article_id as it's a serial number which should be unique